src/go/k8s/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -886,3 +895,43 @@ func disableConsoleReconciliation(console *vectorzied_v1alpha1.Console) {
 	}
 	console.Annotations[managedAnnotationKey] = NotManaged
 }
+
+func validateHelmRelease(rp *v1alpha1.Redpanda, hr *helmv2beta2.HelmRelease) error {
+	errs := make([]error, 0)
+
+	errReplicaCount := validateHelmReleaseReplicaCount(rp, hr)
+	if errReplicaCount != nil {
+		errs = append(errs, errReplicaCount)
+	}
+
+	return errors.Join(errs...)
+}
+
+func validateHelmReleaseReplicaCount(rp *v1alpha1.Redpanda, hr *helmv2beta2.HelmRelease) error {
+	// First validate if we are scaling down too fast
+	clusterSpec := &v1alpha1.RedpandaClusterSpec{}
+	err := json.Unmarshal(hr.Spec.Values.Raw, clusterSpec)
```

Comment:
If the validation function moves from the current place the helm release might be nil and all other fields this code tries to reach.

Reply:
thats correct, this is made obvious in a comment.